### PR TITLE
User dashboard

### PR DESCRIPTION
**Pull request by ronny2205**
I added a link in the navbar to the dashboard, styled the dashboard (replaced the list of active games with a small chess board for each game), and prevented a user go to a different user's dashboard.

### CODE REVIEW

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -8,6 +8,7 @@
 }
 
 #body {
+    font-family: Corbel, "Lucida Grande", "Lucida Sans Unicode", "Lucida Sans", "DejaVu Sans", "Bitstream Vera Sans", "Liberation Sans", Verdana, "Verdana Ref", sans-serif;
```
Comment (jonathanpike) [inline app/assets/stylesheets/users.scss]: This isn't needed, as it's specified for the entire `body` element. All child elements inherit from `body`.  Same as with line 36.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!
   helper_method :games_won?, :games_lost?, :games_draw?, :games_unfinished?
 
   def show
     @user = User.find(params[:id])
     @games = Game.where('black_player_id = ? OR white_player_id = ?', @user.id, @user.id)
+    render text: 'Not Allowed' unless @user == current_user
```
Comment (jonathanpike) [inline app/controllers/users_controller.rb]: Great idea! :+1:

**Comment by jonathanpike** [comment]
Really nice work, @ronny2205!  I absolutely love the idea of the small chessboards!  Some comments (along with the ones I've already made in-line): 

1. Personally, I think it would be more consistent to have the content in a `booyah-box`, to match with the style of the Find Game and Create Game pages.  

2. The colour of the titles on the page, while they match the red of the chessboard, don't match any of our other styling.  I'm fine with colouring the text, but I think it should be consistent. 

3. With titles of different lengths, the little chessboards are all different sizes. Perhaps move the titles to a different location and put something else inside the chessboard?  I've made a couple of changes and have a screenshot below for your comment: 

Odd sized mini-chessboards: 
screen shot 2015-12-29 at 7 04 08 pm

This is my idea for a solution: 
screen shot 2015-12-29 at 7 26 09 pm

You'll note that the colour of the kings corresponds to the colour of the player in the game.  This is just an idea!  I've also added the booyah box.

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -25,4 +26,31 @@
...
+
+td {
+  font-family: Corbel, "Lucida Grande", "Lucida Sans Unicode", "Lucida Sans", "DejaVu Sans", "Bitstream Vera Sans", "Liberation Sans", Verdana, "Verdana Ref", sans-serif;
+  height:45px;
+  text-align:center;
+  vertical-align:middle;
+  width:45px;
+}
+
+.clear {
```
Comment (jonathanpike) [inline app/assets/stylesheets/users.scss]: This rule doesn't seem to be used anywhere.  Do we need it?

Comment (ronny2205) [inline app/assets/stylesheets/users.scss]: The rule on line 43 is not used, you are right! I forgot to delete it.

**Comment by ronny2205** [comment]
@jonathanpike: your solution for the small chessboards is great. Let's use it. 
About the colors for titles, I agree. They should be consistent. I like the reddish color that match the chessboard, but it can be black as well. What do you think is better?

File: app/assets/stylesheets/users.scss (hunk 1 above)
Comment (ronny2205) [inline app/assets/stylesheets/users.scss]: You are right!

**Comment by jonathanpike** [comment]
@ronny2205:  I'm good with the red colour for certain elements.  What about for the Scorecard title and text (ie. Wins, Losses, Draws)?  I think it would work well there.  And Perhaps we can colour the background of the scorecard like the chessboard?  

I asked my wife what she thought about the little chessboards, and she had the following comments: 

1. The title needs to be bigger than I had it.
2. She wasn't sure what the king was for.  I suggest adding a [tooltip](http://getbootstrap.com/javascript/#tooltips) with a message that says "You are playing white" or something.  

Do you want me to push the changes that I made to this branch so you don't have to re-do them?

**Comment by ronny2205** [comment]
@jonathanpike: it's a good idea to color the scoreboard as well. I'll do that and the other suggestions too. (I'll look into the tooltip). Thanks for all the suggestions!
It would be great if you push the changes that you already made. Thanks!

**Comment by jonathanpike** [comment]
@ronny2205: I'll push the changes that I made when I get home today!

**Comment by ronny2205** [comment]
@jonathanpike: Thanks, no rush :)

**Comment by ronny2205** [comment]
Hi,
I added more colors to the dashbord, game names are bigger and I added the tooltips. I think it's a great idea @jonathanpike. The Js code is currently in the user show page. I did try it in the users.js, but it didn't recognize the bootstrap tooltips. It used the html tooltips instead (which are not bad either). It could be related to the time the page is loaded, but I'm not sure.

**Comment by jonathanpike** [comment]
Just pulled the branch down and it looks great @ronny2205.  I love the styling you did with the user scorecard.  I made a few small changes: 

1. I figured out how to move the tooltips to users.js.  It works as expected. 

2. I removed the red colour from the main page titles.  From what we talked about earlier, we wanted consistency.  Since every other page has the main titles in black, I figured that they should be black here. 

If you think those changes are good, I'll go ahead and merge.

**Comment by ronny2205** [comment]
Thanks for those changes @jonathanpike. You can merget it. 
What was the solution for the tooltips? I saw you deleted the users.cofee file.

**Comment by jonathanpike** [comment]
@ronny2205: The solution was to change the file from being a `.coffee` file to a `.js` file.  We aren't writing Coffeescript, and regular Javascript doesn't seem to work well in Coffeescript files.

**Comment by ronny2205** [comment]
@jonathanpike: Interesting. I didn't use the .coffee file. I created a .js file (which I later deleted), but it didn't work.

**Comment by jonathanpike** [comment]
@ronny2205: Maybe you didn't name it quite right? I just renamed the `.coffee` file.

**Pull request merged**